bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -1523,10 +1597,108 @@ func (gcp *Gcp) createSecrets() error {
 	return nil
 }
 
+// setGcpPluginDefaults sets the GcpPlugin defaults.
+func (gcp *Gcp) setGcpPluginDefaults() error {
+	// Set the email
+	if gcp.kfDef.Spec.Email == "" && gcp.gcpAccountGetter != nil {
+		email, err := gcp.gcpAccountGetter()
+		if err != nil {
+			log.Errorf("cannot get gcloud account email. Error: %v", err)
+			return err
+		}
+		email = strings.TrimSpace(email)
+		gcp.kfDef.Spec.Email = email
+	} else {
+		log.Warnf("gcpAccountGetter not set; can't get default email")
+	}
+
+	// Set the defaults that will be used if not explicitly set.
+	// If the plugin is provided these values will be overwritten,
+	pluginSpec := &GcpPluginSpec{}
+	err := gcp.kfDef.GetPluginSpec(GcpPluginName, pluginSpec)
+
+	if err != nil && !kfdefs.IsPluginNotFound(err) {
+		log.Errorf("There was a problem getting the gcp plugin %v", err)
+		return errors.WithStack(err)
+	}
+
+	if pluginSpec.CreatePipelinePersistentStorage == nil {
+		pluginSpec.CreatePipelinePersistentStorage = proto.Bool(pluginSpec.GetCreatePipelinePersistentStorage())
+		log.Infof("CreatePipelinePersistentStorage not set defaulting to %v", *pluginSpec.CreatePipelinePersistentStorage)
+	}
+
+	if pluginSpec.Auth == nil {
+		pluginSpec.Auth = &Auth{}
+	}
+
+	if pluginSpec.Auth.BasicAuth == nil && pluginSpec.Auth.IAP == nil {
+		log.Warnf("Backfilling auth; this is deprecated; Auth should be explicitly set in Gcp plugin")
```

Comment:
when should auth be set in the future, init?

Reply:
I think it should be set in the YAML file.
e.g.
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_gcp_iap.0.6.yaml

So GcpPluginSpec should be specified in that file

Reply:
I see, thanks